server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -70,6 +74,87 @@ protected boolean addMockInternalEngine() {
         return false;
     }
 
+    public void testPrimaryStopped_ReplicaPromoted() throws Exception {
+        final String nodeA = internalCluster().startNode();
+        final String nodeB = internalCluster().startNode();
+        createIndex(INDEX_NAME);
+        ensureGreen(INDEX_NAME);
+
+        client().prepareIndex(INDEX_NAME).setId("1").setSource("foo", "bar").setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).get();
+        refresh(INDEX_NAME);
+
+        waitForReplicaUpdate();
+        assertHitCount(client(nodeA).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), 1);
+        assertHitCount(client(nodeB).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), 1);
+
+        final DiscoveryNode primaryDiscoveryNode = getNodeContainingPrimaryShard();
```

Comment:
nit: If `primaryDiscoveryNode` needs to identified, we can start with only one node. This will prevent another call to identify the primary shard node.

```
final String nodeA = internalCluster().startNode(); --> node with primary shard
        createIndex(INDEX_NAME);
        ensureGreen(INDEX_NAME);

// nodeA contains the primary shard, now start second node which will contain the replica shard. Going forward, we can use nodeA as node containing primary shards.
final String nodeB = internalCluster().startNode();
```

Applicable for this and remaining tests.